tests/test_lab_train.py

```diff
@@ -208,6 +210,86 @@ def test_invalid_data_dir(self, cli_runner: CliRunner):
             assert "Could not read from data directory" in result.output
             assert result.exit_code == 1
 
+    @patch("instructlab.utils.is_macos_with_m_chip", return_value=False)
```

Comment:
Opinion: it would be better to target the new functions in `utils.py` directly with a test - we would not need any mocks, nor we would need to trigger `ilab` through CLI, while still covering the cases we care about - the functions are pure and easy to test without any mocks.

Reply:
While this is definitely an approach which can work well for stable functions, much of this CLI will change over time, so adding fine-grained tests like these will tightly couple us to the functionality of what are essentially scaffolding utils. Therefore, it makes the most sense for the time being to focus on functional and end-to-end tests which allow us to confirm the behavior of what we're writing. 

However; if others feel strongly that this should be a blocker for 0.18.0, we can definitely spend time on this.